src/main/java/software/aws/mcs/auth/SigV4AuthProvider.java

```diff
@@ -373,4 +385,42 @@ static int indexOf(byte[] target, byte[] pattern) {
         // Loop exhaustion means we did not find it
         return -1;
     }
+
+
+    /**
+     * Creates a STS role credential provider
+     * @param roleArn The ARN of the role to assume
+     * @param stsRegion The region of the STS endpoint
+     * @return The STS role credential provider
+     */
+    private static StsAssumeRoleCredentialsProvider createSTSRoleCredentialProvider(String roleArn,
+                                                                     String stsRegion) {
+        //Get role name from ARN
+        String[] arnParts = roleArn.split("/");
+        if(arnParts.length < 2){
+            throw new IllegalArgumentException("Invalid role ARN");
+        }
+        String roleName = arnParts[arnParts.length - 1];
```

Comment:
Would you mind splitting this out into a helper static method and adding a unit test for it (with both happy and sad paths)? Also, should we add a `@Nonnull` to the `roleArn` and `stsRegion` parameters?